repl/src/main/scala/org/apache/spark/repl/ExecutorClassLoader.scala

```diff
@@ -115,23 +127,31 @@ class ExecutorClassLoader(
     }
   }
 
+  // See org.apache.spark.network.server.TransportRequestHandler.processStreamRequest.
+  private val STREAM_NOT_FOUND_REGEX = s"Stream '.*' was not found.".r.pattern
+
   private def getClassFileInputStreamFromSparkRPC(path: String): InputStream = {
-    val channel = env.rpcEnv.openChannel(s"$classUri/$path")
+    val channel = env.rpcEnv.openChannel(s"$classUri/${urlEncode(path)}")
```

Comment:
This fixes an issue in the current ExecutorClassLoader: it doesn't encode the path.

Before this PR, when we try to fetch some class that contains special characters, it will always throw `URISyntaxException: Illegal character in path ...` which will be converted to `ClassNotFoundException` at last.

Reply:
I would be good to cover this with a test.

Reply:
It's covered by `create encoder in executors`. Reverting this line would cause this test fail..